update

Eggsy07 opened the pull request:
I created a basic tutorial for the game player. please check it and let me know your point of view.

Created Tutorials

Eggsy07 committed:
Create TileMap5.tscn

last updates

updating tutorials

update

I created a basic tutorial for the game player. please check it and let me know your point of view.

Created Tutorials

update

Filaipus commented:
Ok eu ainda não olhei para o código (nem testei com um comando) mas num teste muito rápido o que eu diria é:
- O tutorial está a ocupar demasiado do ecrã.
- Nao precisas de dizer "Kill all enemies to complete the level". Visto que é o que o jogador vai fazer e vê o nível a avançar, é algo que ele aprende ao jogar e não é preciso usar palavras :) 
- Esta a faltar uma explicação de como a teleportação funciona. O nível ideal para explicar seria o nível 5, onde não é possível chegar ao inimigo sem saltar para baixo. 
- Tu fizeste uma coisa interessante onde visto que temos várias opções para movimentar a personagem, estás a mostrar tudo o que é possível. Apesar de isso ser uma boa ideia, não sei se funciona muito bem neste momento visto que ocupa MUITO espaço. Eu recomendava fazer só as teclas principais (setas). 
### Se gostas desse sistema (que eu acho que é bastante inteligente), vê como este jogo fez  **[https://kultisti.itch.io/sulka CLICA AQUI]( https://kultisti.itch.io/sulka)** 
É bastante mais simples nos olhos, e o jogador consegue facilmente perceber o que é suposto fazer :)

Filaipus commented:
Ok Jaime eu vi as mudanças. Dois problemas:

O tutorial continua a ocupar imenso espaço no ecrã. 
<img width="289" alt="image" src="https://github.com/GamesForGood/CandyWrapperDeluxe/assets/6901771/e1997807-9a5f-49a4-beca-11c19d8c2c16">
Por exemplo, consegues ver onde o jogador está? 


E eu comecei o jogo com teclado mas está a mostrar as teclas de comando.


Eu recomendava de novo veres este jogo para veres como eles fizeram o que tu queres fazer: https://kultisti.itch.io/sulka

Mas este primeiro ponto é mais uma questão de estética que te estou a tentar ajudar a perceberes. Lembra-te, o propósito deste jogo é para VOCÊS poderem tentar vende-lo. Se estiver muito feio, ninguém vai querer jogar ): 
Eu acho que a animação que meteste nos botões funciona bastante bem, e os botões não são feios. O problema é o tamanho e onde estás a colocar as coisas :)

Filaipus commented:
Jaime encontrei uma seta no pack que estás a usar (Xelu_Free_Controller&Key_Prompts\Others\Arrows)

Eu acho que isto é melhor do que o texto, o que achas? (quanto menos texto tivermos, melhor. Assim pessoas que não falam inglês podem jogar o jogo)
<img width="283" alt="image" src="https://github.com/GamesForGood/CandyWrapperDeluxe/assets/6901771/d8e559b0-92e9-4a36-a707-ee96bfdf6546">

Filaipus reviewed (changes requested):
Jaime, isto são as coisas que tens que concertar antes de poder aceitar:

1. Apagaste o nível 5 por alguma razão
2. O primeiro nível não mostra o tutorial quando estás a usar um teclado

Corrige isso e depois antes de fazeres o upload verifica que o jogo funciona todo bem :)

Filaipus committed:
Update Icons

Unfortunately the scale functions on Godot aren't very friendly to reducing to such small sizes. I had to redraw by hand a lot of the icons, with dubious quality. Either way, it is what it is. They are still a little too big but we can tackle that later.

Filaipus committed:
Change Start Menu Label to "Enter"

For some reason we had the message say "space" instead of "enter", which makes much more sense. Space restricts future tasks such as being able to control the player in the main menu.

Filaipus committed:
Change Input Map

Changed the input map to things that made more sense to me. Also organized the keyboard values, so we can rotate over them in the tutorial. Hopefully once we enable customized controls this is a blessing and not a curse.

Filaipus committed:
Updated Tutorial

This commit is multifold, but it basically cleaned up a lot of the animations/graphics and then logic of the tutorial. This now correctly displays the tutorial over the first 5 stages, with support for keyboard and controller. Some logic had to be added to Tutorials.gd to enable/disable and rotate the possible controls (including some code to NOT show the controls on stage 5).

Not the prettiest code but it works.
